apps/vscode/src/host/hooks.ts

```diff
@@ -186,3 +195,39 @@ async function getStatementRange(
   );
   return { range: unadjustedRange(language, result.range), code: result.code };
 }
+
+class EmbeddedHelpTopicProvider implements HostHelpTopicProvider {
+  private readonly _engine: MarkdownEngine;
+
+  constructor(
+    readonly engine: MarkdownEngine,
+  ) {
+    this._engine = engine;
+  }
+
+  async provideHelpTopic(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+    token: vscode.CancellationToken): Promise<string | undefined> {
+    const vdoc = await virtualDoc(document, position, this._engine);
```

Comment:
There is a lot of boilerplate here for the `virtualDoc` if you compare this to what we do for `provideStatementRange()`; it might be interesting in the longer term to see if we can do anything to abstract that away.